salt/spm/__init__.py

```diff
@@ -245,6 +246,50 @@ def _local_install(self, args, pkg_name=None):
 
         formula_tar.close()
 
+    def _resolve_deps(self, formula_def):
+        '''
+        Return a list of packages which need to be installed, to resolve all
+        dependencies
+        '''
+        pkg_info = self.pkgdb['{0}.info'.format(self.db_prov)](formula_def['name'])
+        if not isinstance(pkg_info, dict):
+            pkg_info = {}
+
+        can_has = {}
+        cant_has = []
+        for dep in formula_def.get('dependencies', '').split(','):
+            dep = dep.strip()
+            if not dep:
+                continue
+            if self.pkgdb['{0}.info'.format(self.db_prov)](dep):
+                continue
+
+            if dep in self.avail_pkgs:
+                can_has[dep] = self.avail_pkgs[dep]
+            else:
+                cant_has.append(dep)
+
+        inspected = []
+        to_inspect = can_has.copy()
+        while len(to_inspect) > 0:
+            dep = to_inspect.keys()[0]
+            del to_inspect[dep]
+
+            # Don't try to resolve the same package more than once
+            if dep in inspected:
+                continue
+            inspected.append(dep)
+
+            repo_contents = self.repo_metadata.get(can_has[dep], {})
+            repo_packages = repo_contents.get('packages', {})
+            dep_formula = repo_packages.get(dep, {}).get('info', {})
+
+            also_can, also_cant = self._resolve_deps(dep_formula)
+            can_has.update(also_can)
+            cant_has = sorted(set(cant_has + also_cant))
```

Comment:
consider also using set's, specially because you're casting them to a set, and ordering, so, order also does not mater, and then:
```python
cant_has = sorted(set_a.union(set_b))
```